core/services/relay/evm/contract_transmitter.go

```diff
@@ -64,6 +73,9 @@ func NewOCRContractTransmitter(
 	if err != nil {
 		return nil, err
 	}
+	if reportToEthTxMeta == nil {
+		reportToEthTxMeta = reportToEthTxMetaNoop
```

Comment:
If this case is only used in tests, can we remove the check, make it required and pass in the no-op func in the test code?

Reply:
There will be cases that injects a real function, currently you can only see one case in tests.